### PR TITLE
PCI-609: Create order following payment

### PR DESCRIPTION
**Pull request by VaughanJackson123**

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/orders/api/service/OrderService.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        this.repository = repository;
+    }
+
+    /**
+     * Used to create an order from a checkout object once payment has been successful.
+     * @param checkout the user's checkout object
+     * @return the resulting order
+     */
+    public Order createOrder(final Checkout checkout) {
+        final Order mappedOrder = mapper.checkoutToOrder(checkout);
```
Comment (rjadhav-ch) [inline src/main/java/uk/gov/companieshouse/orders/api/service/OrderService.java]: Should there be a check here to see if an order with the same `reference` already exists in the db?

Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/orders/api/service/OrderService.java]: Good question. I'll look into that.

File: src/main/java/uk/gov/companieshouse/orders/api/model/AbstractOrderData.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    private List<Item> items = new ArrayList<>();
+
+    private String kind;
+
+    private Links links;
+
+    private String totalBasketCost;
+
+    private PaymentStatus status;
```
Comment (nabeelamirch) [inline src/main/java/uk/gov/companieshouse/orders/api/model/AbstractOrderData.java]: I believe the payment status only appears on the Checkout object

Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/orders/api/model/AbstractOrderData.java]: Good spot! Will fix that.

**Pull request merged**